friends/src/App.js

```diff
@@ -31,18 +42,124 @@ class App extends Component {
     })
   }
 
+  firstUpdate = (e, stuff) => {
+    this.setState({
+      updFriend: stuff
+    })
+  }
+
+  handleChanges = e => {
+    this.setState({
+      newFriend: {
+        ...this.state.newFriend,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  handleChanges2 = e => {
+    this.setState({
+      updFriend: {
+        ...this.state.updFriend,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  addFriend = event => {
+    event.preventDefault();
+    console.log('adding friendo here')
+    
+    axios
+    .post('http://localhost:5000/friends', this.state.newFriend)
+    .then( res => {
+      console.log(res)
+      
+    })
+    .catch( err => {
+      console.log(err)
+    })
+
+  }
+
+  deleteFriend = (event, deadFriend) => {
+    
+    console.log(`Deleting ${deadFriend.name} here`)
+
+    axios
+    .delete(`http://localhost:5000/friends/${deadFriend.id}`)
+    .then (res => {
+      console.log(res)
+      
```

Comment:
`this.setState({friends: res.data})` to get that re-render without a window refresh